src/wp-includes/fonts/class-wp-font-face.php

```diff
@@ -389,6 +379,58 @@ private function build_font_face_css( array $font_face ) {
 		return $css;
 	}
 
+	/**
+	 * Normalizes a font-face name for use in CSS.
+	 *
+	 * Add quotes to the font-face name and escape problematic characters.
+	 *
+	 * @see https://www.w3.org/TR/css-fonts-4/#font-family-desc
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param string $font_family The font-face name to normalize.
+	 * @return string The normalized font-face name.
+	 */
+	protected function normalize_css_font_family( string $font_family ): string {
+		$font_family = trim( $font_family, " \t\r\f\n" );
+
+		if (
+			strlen( $font_family ) > 1 &&
+			( '"' === $font_family[0] && '"' === $font_family[ strlen( $font_family ) - 1 ] ) ||
+			( "'" === $font_family[0] && "'" === $font_family[ strlen( $font_family ) - 1 ] )
+		) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'Font family should not be wrapped in quotes; they will be added automatically.' ),
+				'6.9.0'
+			);
+			return $font_family;
+		}
+
+		return '"' . strtr(
+			$font_family,
+			array(
+				/*
+				 * Normalize preprocessed whitespace.
+				 * https://www.w3.org/TR/css-syntax-3/#input-preprocessing
+				 */
+				"\r"   => '\\A ',
+				"\f"   => '\\A ',
+				"\r\n" => '\\A ',
+
+				/*
+				 * CSS Unicode escaping for problematic characters.
+				 * https://www.w3.org/TR/css-syntax-3/#escaping
+				 */
+				"\n"   => '\\A ',
+				'\\'   => '\\5C ',
+				','    => '\\2C ',
```

Comment:
Is this one necessary? Not that it's wrong, but we're producing a quoted value anyway. It shouldn't derail any syntax parsing and, within a string, it's seen as a comma either way.

Reply:
When printing CSS, the font system checks for `,` and does a split/join in an effort to normalize the font family names:

https://github.com/WordPress/wordpress-develop/blob/4d43703432cfc88ea6ae59c568cd70a22f993a0c/src/wp-includes/fonts/class-wp-font-utils.php#L67-L76

It seemed preferable to escape `,` in the single font family name here to try to avoid being mangled later.

Reply:
Ah so it's a workaround for the deficiencies of another part of the system. Could we go the other way around and parse the value instead of exploding by `,`? The CSS Processor has some code that could help with this.

Reply:
It doesn't today, but it could. [CSS Processor](https://github.com/WordPress/php-toolkit/blob/trunk/components/DataLiberation/CSS/class-cssprocessor.php) is just a tokenizer, not a full-blown parser. Or you might want to reuse just the part of it that parses strings, just to avoid the blanket `explode()` call.